scala/lambdas/preingest-tdr-package-builder/src/test/scala/uk/gov/nationalarchives/preingesttdrpackagebuilder/LambdaTest.scala

```diff
@@ -104,13 +104,15 @@ class LambdaTest extends AnyFlatSpec with ScalaCheckDrivenPropertyChecks:
 
       def stripFileExtension(title: String) = if title.contains(".") then title.substring(0, title.lastIndexOf('.')) else title
 
-      val metadataObjects = s3Contents(s"${testData.batchId}/metadata.json").asInstanceOf[List[MetadataObject]]
-      val contentFolderMetadataObject = metadataObjects.head.asInstanceOf[ContentFolderMetadataObject]
-      val assetMetadataObject = metadataObjects(1).asInstanceOf[AssetMetadataObject]
-      val fileMetadataObject = metadataObjects(2).asInstanceOf[FileMetadataObject]
-      val metadataFileMetadataObject = metadataObjects(3).asInstanceOf[FileMetadataObject]
+      val metadataObjects: List[MetadataObject] = s3Contents(s"${testData.batchId}/metadata.json").asInstanceOf[List[MetadataObject]]
+      val contentFolderMetadataObject = metadataObjects.find(_.isInstanceOf[ContentFolderMetadataObject]).get.asInstanceOf[ContentFolderMetadataObject]
```

Comment:
If you want, you could do `val metadataObjectsGrouped = metadataObjects.groupBy(_.getClass.getSimpleName)` and then for each

```
val contentFolderMetadataObject = metadataObjectsGrouped("ContentFolderMetadataObject").head
val assetMetadataObject = metadataObjectsGrouped("AssetFolderMetadataObject").head
val fileMetadataObjects = metadataObjectsGrouped("FileFolderMetadataObject")
```

Alternatively, you could do

```
val contentFolderMetadataObject = metadataObjects.collect { case contentFolder: ContentFolderMetadataObject => contentFolder }.head
```

Reply:
I like the `collect` one. I've done that for each of them.